lib/auth_web/controllers/api_controller.ex

```diff
@@ -48,4 +48,24 @@ defmodule AuthWeb.ApiController do
         json(conn, roles)
     end
   end
+
+  @doc """
+  `end_session/2` logs the person out of their session.
+  """
+  def end_session(conn, %{"client_id" => client_id, "session_id" => session_id}) do
+
+    case Auth.Apikey.decode_decrypt(client_id) do
+      {:error, _} ->
+        Auth.Log.error(conn, %{client_id: client_id})
+        AuthWeb.AuthController.unauthorized(conn)
+
+      # pfovided the client_id successfully decrypted, we can end the session:
+      {:ok, _app_id} ->
+        conn
+        # a REST API request will not have the conn.assigns.sid so set it here:
+        |> assign(:sid, session_id)
+        |> Auth.Session.end_session()
```

Comment:
Here the `sid` is added to `conn` so the `end_session` function can update the `end` session value. The `sid` is then removed from the `conn`.
Instead of passing `conn` to the function I think we can pass the `session_id` directly. We'll need to udpate also the parameter for `end_session` and `update_session_end` but I think it makes things a be more easy to read and we'll be able to remove `|> assign(:sid, session_id)`

Reply:
That seems like a sensible refactor. The only reason I was doing adding the `sid`  to `conn` was for pipeline. But if instead we make the `session_id` the second parameter of `Auth.Session.end_session` then it will be much cleaner. 💭

Reply:
I think we can even remove completely the `conn` and just have one argument. I'll check this on a new branch